gcc/rust/expand/rust-macro-invoc-lexer.h

```diff
@@ -22,6 +22,33 @@
 #include "rust-ast.h"
 
 namespace Rust {
+template <class T> class MacroInvocLexerBase
+{
+public:
+  MacroInvocLexerBase (std::vector<T> stream)
+    : offs (0), token_stream (std::move (stream))
+  {}
+
+  // Advances current token to n + 1 tokens ahead of current position.
+  void skip_token (int n) { offs += (n + 1); }
+
+  // Skips the current token.
+  void skip_token () { skip_token (0); }
+
+  std::string get_filename () const
+  {
+    // FIXME
+    gcc_unreachable ();
+    return "FIXME";
+  }
```

Comment:
Do we need that function in the base class?

Reply:
Whatever mechanism we end up using for ```get_filename``` would probably (?) involve an extra ```filename``` field, which would be handled the same way for all child classes